dietpi/dietpi-software

```diff
@@ -11652,6 +11659,43 @@ _EOF_
 			G_EXEC chmod +x /usr/local/bin/restic
 		fi
 
+		if To_Install 141 # MediaWiki
+		then
+			# Install required PHP modules: https://www.mediawiki.org/wiki/Manual:Installation_requirements#PHP
+			aDEPS=("php$PHP_VERSION-intl")
+			# - Add JSON module for PHP7, as it does not exist (embedded in core package) on PHP8
+			local json=()
+			[[ $PHP_VERSION == 8* ]] || aDEPS+=("php$PHP_VERSION-json") json=('json')
+
+			# Create MariaDB database and user
+			if [[ -d '/mnt/dietpi_userdata/mysql/mediawiki' ]]
+			then
+				G_DIETPI-NOTIFY 2 'MediaWiki MariaDB database found, will NOT overwrite.'
+			else
+				/boot/dietpi/func/create_mysql_db mediawiki mediawiki "$GLOBAL_PW"
+			fi
+
+			# Get latest version
+			local version="1.39"
+			local file="mediawiki-1.39.3"
+			local fallback_url="https://releases.wikimedia.org/mediawiki/1.39/mediawiki-1.39.3.tar.gz"
+			Download_Install "https://releases.wikimedia.org/mediawiki/$version/$file.tar.gz"
```

Comment:
```suggestion
			local version='1.39.3'
			Download_Install "https://releases.wikimedia.org/mediawiki/${version%.*}/mediawiki-$version.tar.gz"
```
- Probably we can at least get the latest patch version from the downloads, so we only need to regularly check for new minor versions: https://releases.wikimedia.org/mediawiki/1.39/
- Not sure whether it is safe to assume the second latest minor version to be the latest stable one? https://releases.wikimedia.org/mediawiki/
- We can check once v1.40 has been released whether there is a `/1.41/` directory immediately.
- Last but not least we could scrape the latest version from the main download page: https://www.mediawiki.org/wiki/Download

Reply:
Yes, I used the hardcode values for testing. I was hoping for some clever ideas from your side on how best to retrieve the correct version. Good idea to accept the second newest minor version as the newest stable version.

Reply:
I will have a look about whether it can be well obtained from the main download page. Otherwise I think we should wait for v1.40 release to see whether a v1.41 directory really already exists at this point.

I mean we could also check the newest first, see whether there is a download listed without `-rc` or any such suffix, else check next dir. But IMO too complicated, requiring too many network requests etc.